src/lib/transition_frontier/transition_frontier.ml

```diff
@@ -748,62 +749,69 @@ module Make (Inputs : Inputs_intf) :
         (* 4 *)
         (* note: new_root_node is the same as root_node if the root didn't change *)
         let garbage_breadcrumbs, new_root_node =
-          if distance_to_parent > max_length then (
-            Logger.info t.logger ~module_:__MODULE__ ~location:__LOC__
-              !"Distance to parent: %d exceeded max_lenth %d"
-              distance_to_parent max_length ;
+          if distance_to_root > max_length then (
+            Logger.debug t.logger ~module_:__MODULE__ ~location:__LOC__
+              "Moving the root of the transition frontier. The new node is \
+               $distance_to_root blocks after the root, which exceeds the max \
+               length of $max_length."
+              ~metadata:
+                [ ("distance_to_parent", `Int distance_to_root)
+                ; ("max_length", `Int max_length) ] ;
             (* 4.I *)
             let heir_hash = List.hd_exn (hash_path t node.breadcrumb) in
             let heir_node = Hashtbl.find_exn t.table heir_hash in
             (* 4.II *)
-            let bad_hashes =
+            let bad_children_hashes =
               List.filter root_node.successor_hashes
                 ~f:(Fn.compose not (State_hash.equal heir_hash))
             in
-            let bad_nodes =
-              List.map bad_hashes ~f:(Hashtbl.find_exn t.table)
+            let garbage_hashes =
+              bad_children_hashes
+              @ List.bind bad_children_hashes ~f:(successor_hashes_rec t)
```

Comment:
this is the key bit, to get all the successors, yes?

Reply:
yeah